pallets/fusion/src/lib.rs

```diff
@@ -0,0 +1,3044 @@
+		Twox64Concat,
+		CurrencyId,
+		FusionUserCurrencyBalance,
+		OptionQuery,
+	>;
+
+	/// Stores era rewards for each pool
+	#[pallet::storage]
+	#[pallet::getter(fn fusion_era_rewards)]
+	pub type FusionEraRewards<T: Config> = StorageDoubleMap<
+		_,
+		Twox64Concat,
+		EraIndex,
+		Twox64Concat,
+		PoolId,
+		EraReward<T>,
+		OptionQuery,
+	>;
+
+	/// Stores the conversion rates for currencies
+	/// How much one unit of currency is equal in AVAIL
+	#[pallet::storage]
+	#[pallet::getter(fn fusion_currency_rates)]
+	pub type FusionCurrencyRates<T: Config> = StorageDoubleMap<
+		_,
+		Twox64Concat,
+		EraIndex,
+		Twox64Concat,
+		CurrencyId,
+		BalanceOf<T>,
+		OptionQuery,
+	>;
+
+	/// Stores the next currency changes to be applied next era
+	#[pallet::storage]
+	#[pallet::getter(fn fusion_currency_rate_changes)]
+	pub type FusionCurrencyRateChanges<T: Config> =
+		StorageMap<_, Twox64Concat, CurrencyId, BalanceOf<T>, OptionQuery>;
+
+	/// Mapping from EVM Address to Substrate address
+	#[pallet::storage]
+	#[pallet::getter(fn fusion_evm_to_substrate_address)]
+	pub type FusionEVMToSubstrateAddress<T: Config> =
+		StorageMap<_, Blake2_128Concat, EvmAddress, T::AccountId, OptionQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn era_durations)]
+	pub type EraDurations<T: Config> = StorageMap<_, Twox64Concat, EraIndex, u64, OptionQuery>;
+
+	/// Stores the fusion era exposure for HistoryDepth eras
+	#[pallet::storage]
+	#[pallet::getter(fn fusion_era_data)]
+	pub type FusionExposures<T: Config> = StorageDoubleMap<
+		_,
+		Twox64Concat,
+		EraIndex,
+		Twox64Concat,
+		PoolId,
+		FusionExposure<T>,
+		OptionQuery,
+	>;
+
+	/// Store the pools that backed the validator set as the key
+	#[pallet::storage]
+	#[pallet::getter(fn fusion_pools_from_validator)]
+	pub type FusionPoolsFromValidator<T: Config> = StorageDoubleMap<
+		_,
+		Twox64Concat,
+		EraIndex,
+		Twox64Concat,
+		T::AccountId,
+		BoundedVec<PoolId, T::MaxPoolsPerValidator>,
+		ValueQuery,
+	>;
+
+	/// Stores the fusion claimed rewards for HistoryDepth eras
+	#[pallet::storage]
+	#[pallet::getter(fn claimed_rewards)]
+	pub type ClaimedRewards<T: Config> = StorageDoubleMap<
+		_,
+		Twox64Concat,
+		EraIndex,
+		Twox64Concat,
+		(PoolId, EvmAddress),
+		BalanceOf<T>,
+		ValueQuery,
+	>;
+
+	/// Stores EVM Address of the slash destination
+	/// It can be controlled with technical committee
+	#[pallet::storage]
+	#[pallet::getter(fn slash_destination)]
+	pub type SlashDestination<T> = StorageValue<_, EvmAddress, OptionQuery>;
+
+	/// Stores true if for a given era, a validator and a pool funds account, a slash is pending
+	/// (era, (validator, funds_account)) => has_pending_slash
+	/// Used mainly to quickly determine if a slashed nominator is from Fusion pallet
+	#[pallet::storage]
+	#[pallet::getter(fn has_pending_slash)]
+	pub type HasPendingSlash<T: Config> = StorageDoubleMap<
+		_,
+		Twox64Concat,
+		EraIndex,
+		Twox64Concat,
+		(T::AccountId, T::AccountId),
+		bool,
```

Comment:
Having an entry for the pool account itself indicates the pool's slashing exposure. Instead of bool, we can store the slashable balance, similar to the info in the `ValidatorSlashInEra` storage of the staking pallet.